lib/src/Base/Stat/CovarianceModelImplementation.cxx

```diff
@@ -947,7 +951,12 @@ void CovarianceModelImplementation::setFullParameter(const Point & parameter)
     scale_[i] = parameter[index];
     ++ index;
   }
-  // Second the amplitude parameter
+  // Second the nugget factor
```

Comment:
in order to not break the current API, I will rather sugget to put this block after the specific one for `amplitude`

Reply:
You know, this is actually what I did at first, but I decided against it in the end, because when you read the expression of the covariance matrix:

$$
\sigma² (\boldsymbol{R}_\theta + \eta \boldsymbol{I}_n)
$$

you want to : 
1. determine the input correlation matrix $\boldsymbol{R}_\theta$ (parametrized by the scale parameter), then 
2. the $\eta$ scalar which is added to its diagonal elements (the nuggetFactor), 
3. and then the amplitude $\sigma$ whose square is applied to the resulting augmented "correlation" matrix $\boldsymbol{R}_\theta + \eta \boldsymbol{I}_n$.

Furthermore, note that `CorrelationMatrixImplementation` actually has additional parameters after the scale and the amplitude: the nondiagonal elements of the **output** correlation matrix. So, IMO, it is more logical to progress from the parameters which affect the **input** correlation matrix (scale and nuggetFactor) to the parameters which affect the output covariance matrix (the amplitude values and the coefficients of the output correlation matrix).

Reply:
I think this ordering point is less important now this PR includes easy-to-use methods to activate/deactivate parameters: `activateScale`, `activateNuggetFactor`, `activateAmplitude`.